src/server/restOICP/v2.3.0/service/OICPEndpointService.ts

```diff
@@ -0,0 +1,451 @@
+import { StatusCodes } from 'http-status-codes';
+import TenantComponents from '../../../../types/TenantComponents';
+import TenantStorage from '../../../../storage/mongodb/TenantStorage';
+import Utils from '../../../../utils/Utils';
+import UtilsService from '../../../rest/v1/service/UtilsService';
+
+const MODULE_NAME = 'OICPEndpointService';
+
+export default class OICPEndpointService {
+  static async handleDeleteOicpEndpoint(action: ServerAction, req: Request, res: Response, next: NextFunction): Promise<void> {
+    // Check if component is active
+    UtilsService.assertComponentIsActiveFromToken(req.user, TenantComponents.OICP,
+      Action.DELETE, Entity.OICP_ENDPOINT, MODULE_NAME, 'handleDeleteOicpEndpoint');
+    // Filter
+    const filteredRequest = OICPEndpointSecurity.filterOicpEndpointDeleteRequest(req.query);
+    UtilsService.assertIdIsProvided(action, filteredRequest.ID, MODULE_NAME, 'handleDeleteOicpEndpoint', req.user);
+    // Check auth
+    if (!Authorizations.canDeleteOicpEndpoint(req.user)) {
+      throw new AppAuthError({
+        errorCode: HTTPAuthError.ERROR,
+        user: req.user,
+        action: Action.DELETE, entity: Entity.OICP_ENDPOINT,
+        module: MODULE_NAME, method: 'handleDeleteOicpEndpoint',
+        value: filteredRequest.ID
+      });
+    }
+    // Get
+    const oicpEndpoint = await OICPEndpointStorage.getOicpEndpoint(req.user.tenantID, filteredRequest.ID);
+    UtilsService.assertObjectExists(action, oicpEndpoint, `OICPEndpoint with ID '${String(filteredRequest.ID)}' does not exist`,
+      MODULE_NAME, 'handleDeleteOicpEndpoint', req.user);
+    // Delete
+    await OICPEndpointStorage.deleteOicpEndpoint(req.user.tenantID, oicpEndpoint.id);
+    // Log
+    Logging.logSecurityInfo({
+      tenantID: req.user.tenantID,
+      user: req.user, module: MODULE_NAME, method: 'handleDeleteOicpEndpoint',
+      message: `Oicp Endpoint '${oicpEndpoint.name}' has been deleted successfully`,
+      action: action,
+      detailedMessages: { oicpEndpoint }
+    });
+    // Ok
+    res.json(Constants.REST_RESPONSE_SUCCESS);
+    next();
+  }
+
+  static async handleCreateOicpEndpoint(action: ServerAction, req: Request, res: Response, next: NextFunction) {
+    // Check auth
+    if (!Authorizations.canCreateOicpEndpoint(req.user)) {
+      throw new AppAuthError({
+        errorCode: HTTPAuthError.ERROR,
+        user: req.user,
+        action: Action.CREATE,
+        entity: Entity.OICP_ENDPOINT,
+        module: MODULE_NAME,
+        method: 'handleCreateOicpEndpoint'
+      });
+    }
+    // Filter
+    const filteredRequest = OICPEndpointSecurity.filterOicpEndpointCreateRequest(req.body);
+    // Check Mandatory fields
+    UtilsService.checkIfOICPEndpointValid(filteredRequest, req);
+    const oicpEndpoint: OICPEndpoint = {
+      ...filteredRequest,
+      createdBy: { id: req.user.id },
+      createdOn: new Date(),
+      status: OICPRegistrationStatus.NEW,
+      version: '2.3.0',
```

Comment:
Please avoid putting magic number directly in the code. 
Create a constant at the level needed to reuse it where needed.

Reply:
done